### DISCUSSION TITLE
Define which impl. should be used

### DISCUSSION THREAD

**Discussion by galanisd** [eclipse-edc/Connector#3427] [General]
`PolicyDefinitionServiceImpl `uses  `PolicyDefinitionStore policyStore`

there are two implementations of `PolicyDefinitionStore `

```
SqlPolicyDefinitionStore extends AbstractSqlStore implements PolicyDefinitionStore 
InMemoryPolicyDefinitionStore implements PolicyDefinitionStore
```

How we define which one should be used at runtime?

This is a general question it applies in many similar cases.

**Comment by galanisd** [comment]
When I added the dependency that contains `SqlPolicyDefinitionStore extends AbstractSqlStore implements PolicyDefinitionStore` 
it prefers it over the other option. However, how we define which one has priority

**Reply by wolf4ood** [reply]
Hi @galanisd  by default there are some bundled implementation mostly in memory and they are flagged as `default` and the impl will be overwritten by another impl if present, check it out [here](https://github.com/eclipse-edc/Connector/blob/main/core/control-plane/control-plane-core/src/main/java/org/eclipse/edc/connector/ControlPlaneDefaultServicesExtension.java#L81)

**Reply by galanisd** [reply]
Ok. But if someone creates another impl

`SqlPolicyDefinitionStore2 extends AbstractSqlStore implements PolicyDefinitionStore` 
Which one will be selected?

**Reply by jimmarino** [reply]
Place whichever implementation you want on the classpath and make sure the other is not on the classpath e.g., modify your runtime image.

As @wolf4ood mentioned, `default` services provide a fallback and are generally used for testing.

**Comment by galanisd** [comment]
Thanks!
